adapter/internal/discovery/xds/server.go

```diff
@@ -583,9 +583,15 @@ func DeleteAPIsWithUUID(vhost, uuid string, environments []string, organizationI
 }
 
 // DeleteAPIWithAPIMEvent deletes API with the given UUID from the given gw environments
-func DeleteAPIWithAPIMEvent(uuid, organizationID string, environments []string) {
+func DeleteAPIWithAPIMEvent(uuid, organizationID string, environments []string, revisionUUID string) {
 	apiIdentifiers := make(map[string]struct{})
 
+	conf, errReadConfig := config.ReadConfigs()
+	if errReadConfig != nil {
+		logger.LoggerOasparser.Fatal("Error loading configuration. ", errReadConfig)
+	}
```

Comment:
we don't need to check this error here. error is only thrown when initial config reading happen from the config file. here its just returning loaded config object.
```suggestion
	conf, _ := config.ReadConfigs()
```

Reply:
Fixed via : https://github.com/wso2/product-microgateway/pull/2503/commits/4d77d188a4aefb79e5d8582c06613d46ae0a1348